worker/src/RTC/RtpPacket.cpp

```diff
@@ -600,12 +598,94 @@ namespace RTC
 
 			auto currentLen = extension->len + 1;
 
-			// Fill with 0's if new length is minor.
-			if (len < currentLen)
-				std::memset(extension->value + len, 0, currentLen - len);
+			MS_DEBUG_DEV(
+			  "set extension id: %" PRIu8 ", length:%" PRIu8 ", currentLen:%" PRIu8 " value:%s",
+			  id,
+			  len,
+			  currentLen,
+			  value.c_str());
+			
+			if (len != currentLen)
+			{
+				// If len != currentLen, means we may need to shift extension or payload.
+				uint8_t* extensionStart    = reinterpret_cast<uint8_t*>(this->headerExtension) + 4;
+				uint8_t* extensionEnd      = extensionStart + GetHeaderExtensionLength();
+				uint8_t* ptr               = extensionStart;
+				size_t extensionsTotalSize = static_cast<size_t>(len + 1);
+				uint8_t* ptr1              = ptr;
```

Comment:
```suggestion
                         	size_t extensionsTotalSize = static_cast<size_t>(len + 1);
				uint8_t* ptr               = extensionStart;
				uint8_t* ptr1              = extensionStart;
```

Reply:
Can you provide a more descriptive names to `ptr` and `ptr1`? How do they differ?

Reply:
ptr will go forward always, ptr1 only go forward with data.  ptr，and movePtr?  may you give me some advice?